### ISSUE TITLE
Question: Can I add my own Metrics?

### ISSUE THREAD

**Issue by luffy06** [tensorflow/tensor2tensor#1270]
### Description

I've added my own problem using `--t2t_usr_dir` flag. And I also want to add my own Metrics for my own problem. But I notice that 

> You can do so for models, hyperparameter sets, modalities, and problems. 

I want to know whether I could add my own Metrics or how I could add my own Metrics ?

Thanks.

### Environment information

```
OS: Ubuntu

$ pip freeze | grep tensor
mesh-tensorflow==0.0.4
tensor2tensor==1.11.0
tensorflow==1.12.0
tensorflow-gpu=1.12.0
tensorflow-hub==0.1.1
tensorflow-metadata==0.9.0
tensorflow-probability=0.5.0
tensorflow-tensorboard=0.4.0

$ python -V
Python 3.5.4 :: Continuum Analytics, Inc.
```

**Comment by luffy06** [comment]
@afrozenator Thanks. I know how to add, but I don't know how to run directly with modified source code.

**Comment by afrozenator** [comment]
Have you tried a commandline like the following?

```
t2t-trainer \
  --generate_data \
  --data_dir=~/t2t_data \
  --output_dir=~/t2t_train/mnist \
  --problem=image_mnist \
  --model=shake_shake \
  --hparams_set=shake_shake_quick \
  --train_steps=1000 \
  --eval_steps=100
```

If you add the metrics in your problem statement, they should run along with it -- does that make sense?

**Comment by afrozenator** [comment]
@deepbrainwaves - ah understood, I don't think that is possible. Also if the metric and the problem maybe useful generally, please consider adding it to the main repo :)

**Comment by luffy06** [comment]
@afrozenator Can I modify the tensor2tensor source code and build it by using `setup.py` rather than `pip install tensor2tensor`？

**Comment by afrozenator** [comment]
Sorry I just saw this and you sent in your first PR as well, thanks!

**Comment by luffy06** [comment]
It's very nice of you, thanks again!
